File: Source/Java/nl/devnology/domain/book/Book.java
package nl.devnology.domain.book;

/**
 * @author Erik Pragt
 */
public class Book {
    private String author;
    private String title;
    private String isbn;
    private Integer price;

    public Book(String author, String title, String isbn, Integer price) {
        this.author = author;
        this.title = title;
        this.isbn = isbn;
        this.price = price;
    }

    public String getTitle() {
        return title;
    }

    public String getAuthor() {
        return author;
    }

    public Integer getPrice() {
        return price;
    }

    public String getIsbn() {
        return isbn;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Book book = (Book) o;

        if (!author.equals(book.author)) return false;
        if (!title.equals(book.title)) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = title.hashCode();
        result = 31 * result + author.hashCode();
        return result;
    }
}

File: Source/C#/DevnologyFitnesseDojo/DevnologyFitnesseDojo/Domain/Books/Book.cs

namespace DevnologyFitnesseDojo.Domain.Books
{
    public class Book
    {
        public string Title { get; set; }
        public string Author { get; set; }
        public string Isbn { get; set; }
        public decimal Price { get; set; }

        public override bool Equals(object o)
        {
            var book = o as Book;

            if (book != null)
            {
                return Title == book.Title && Author == book.Author;
            }

            return false;
        }

        public override int GetHashCode()
        {
            return Title.GetHashCode() + Author.GetHashCode();
        }
    }
}
